File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& a) {
        int n = a.size();
        int maxp = 0;
        int buy = a[0];
        for(int i=0;i<n;i++){
            buy = min(buy,a[i]);
            maxp = max(maxp,a[i]-buy);
        }
        return maxp;
    }
};

File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] a) {
        int n = a.length;
        int ans = 0;
        int buy=a[0];
        
        for(int i=1;i<n;i++){
            if(a[i]<buy){
                buy=a[i];
            }    
            ans = Math.max(ans,a[i]-buy);
        }
        return ans;
    }
}